mongo_orchestration/servers.py

```diff
@@ -47,31 +48,23 @@ def __init_db(self, dbpath):
             os.makedirs(dbpath)
         return dbpath
 
-    def __init_auth_key(self, auth_key, folder):
-        key_file = os.path.join(os.path.join(folder, 'key'))
-        with open(key_file, 'w') as fd:
-            fd.write(auth_key)
-        os.chmod(key_file, stat.S_IRUSR)
-        return key_file
-
     def __init_logpath(self, log_path):
         if log_path and not os.path.exists(os.path.dirname(log_path)):
             os.makedirs(log_path)
 
-    def __init_mongod(self, params, ssl):
+    def __init_mongod(self, params, add_auth=False):
         cfg = self.mongod_default.copy()
         cfg.update(params)
-        cfg.update(ssl)
 
         # create db folder
         cfg['dbpath'] = self.__init_db(cfg.get('dbpath', None))
 
         # use keyFile
-        if self.auth_key:
+        if add_auth and self.auth_key:
             cfg['auth'] = True
-            cfg['keyFile'] = self.__init_auth_key(self.auth_key, cfg['dbpath'])
+            cfg['keyFile'] = self.key_file
 
-        if self.login:
+        if add_auth and self.login:
             cfg['auth'] = True
```

Comment:
Could all of this be:

if add_auth:
    cfg['auth'] = True
    if self.auth_key:
        cfg['keyFile'] = self.key_file